src/js/components/DataTable/Header.js

```diff
@@ -345,7 +345,14 @@ const Header = forwardRef(
               }
 
               // content should fill any available space in cell
-              content = <Box flex="grow">{content}</Box>;
+              content = (
+                <Box
+                  flex="grow"
+                  fill={onSort && sortable !== false ? 'vertical' : false}
```

Comment:
Wouldn't we still want the content to fill even if it's not sortable? For example for cases like: https://storybook.grommet.io/?path=/story/visualizations-datatable-vertical-align--vertical-align

Reply:
Or just a broader question, can we just remove that adjustment all together? Does it work with just the other CSS modifications you've made?

Reply:
Yes I will update this so that fill is always true. 
I'm going to go back looking into the vertical align story more, in this PR when I view the story with the hpe theme it is not respecting the vertical alignment and instead just aligning center. Marking as a draft while I look into this...

Reply:
> Or just a broader question, can we just remove that adjustment all together? Does it work with just the other CSS modifications you've made?

Yeah, was actually able to remove in latest commit

Reply:
Removing `fill` caused an issue with the resizable cells where the cell wasn't stretching when it was resized so I added back the fill prop when the columns are resizable `fill={onResize ? 'vertical' : false}`.